commit 6567515e4a7c42931aefd2c503882839570b7546
Author: Florian Westphal <fw@strlen.de>
Date:   Fri Dec 28 01:24:49 2018 +0100

    netfilter: nf_conncount: fix argument order to find_next_bit
    
    commit a007232066f6839d6f256bab21e825d968f1a163 upstream.
    
    Size and 'next bit' were swapped, this bug could cause worker to
    reschedule itself even if system was idle.
    
    Fixes: 5c789e131cbb9 ("netfilter: nf_conncount: Add list lock and gc worker, and RCU for init tree search")
    Reviewed-by: Shawn Bohrer <sbohrer@cloudflare.com>
    Signed-off-by: Florian Westphal <fw@strlen.de>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/netfilter/nf_conncount.c b/net/netfilter/nf_conncount.c
index f0b05dfebc6e..7554c56b2e63 100644
--- a/net/netfilter/nf_conncount.c
+++ b/net/netfilter/nf_conncount.c
@@ -488,7 +488,7 @@ static void tree_gc_worker(struct work_struct *work)
 	clear_bit(tree, data->pending_trees);
 
 	next_tree = (tree + 1) % CONNCOUNT_SLOTS;
-	next_tree = find_next_bit(data->pending_trees, next_tree, CONNCOUNT_SLOTS);
+	next_tree = find_next_bit(data->pending_trees, CONNCOUNT_SLOTS, next_tree);
 
 	if (next_tree < CONNCOUNT_SLOTS) {
 		data->gc_tree = next_tree;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1012_6567515e4a7c42931aefd2c503882839570b7546.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1012_6567515e4a7c42931aefd2c503882839570b7546.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
